packages/sdk/src/types/blockItemSummary.ts

```diff
@@ -515,14 +527,36 @@ export function affectedAccounts(summary: BlockItemSummary): AccountAddress.Type
             return [summary.removed.account];
         case TransactionKindString.Update: {
             return summary.events.reduce(
-                (addresses: AccountAddress.Type[], event) => {
-                    if (
-                        event.tag === TransactionEventTag.Transferred &&
-                        !addresses.some(AccountAddress.equals.bind(undefined, event.to))
-                    ) {
-                        return [...addresses, event.to];
+                (addresses: Upward<AccountAddress.Type>[], event) => {
+                    if (!isKnown(event)) {
+                        return [...addresses, null];
+                    }
+                    if (event.tag !== TransactionEventTag.Transferred) {
+                        return addresses;
+                    }
+                    return addUnique(addresses, event.to, AccountAddress.equals);
+                },
+                [summary.sender]
+            );
+        }
+        case TransactionKindString.TokenUpdate: {
```

Comment:
In the `rust-sdk` the token events in the `tokenCreation` tx are also considered:

```
   BlockItemSummaryDetails::TokenCreationDetails(token_creation_details) => {
                let mut addresses = BTreeSet::new();
                add_token_event_addresses(&mut addresses, &token_creation_details.events);
                addresses.into_iter().collect()
            }
```